lightning/src/ln/channelmanager.rs

```diff
@@ -4389,6 +4406,42 @@ where
 		Ok(())
 	}
 
+	/// Signals channel readiness after a [`Event::PendingChannelReady`].
+	///
+	/// The `channel_id` parameter indicates which channel should signal readiness,
+	/// and the `counterparty_node_id` parameter is the id of the peer the channel is with.
+	///
+	/// [`Event::PendingChannelReady`]: events::Event::PendingChannelReady
+	pub fn signal_channel_readiness(&self, channel_id: &[u8; 32], counterparty_node_id: &PublicKey) -> Result<(), APIError> {
+		let _persistence_guard = PersistenceNotifierGuard::notify_on_drop(&self.total_consistency_lock, &self.persistence_notifier);
+
+		let per_peer_state = self.per_peer_state.read().unwrap();
+		let peer_state_mutex = per_peer_state.get(counterparty_node_id)
+			.ok_or_else(|| APIError::ChannelUnavailable { err: format!("Can't find a peer matching the passed counterparty node_id {}", counterparty_node_id) })?;
+		let mut peer_state_lock = peer_state_mutex.lock().unwrap();
+		let peer_state = &mut *peer_state_lock;
+		let pending_msg_events = &mut peer_state.pending_msg_events;
+		match peer_state.channel_by_id.entry(channel_id.clone()) {
+			hash_map::Entry::Occupied(mut channel) => {
+				let best_block_height = self.best_block.read().unwrap().height();
+				let channel = channel.get_mut();
+				let (channel_ready_opt, _) = channel.check_get_channel_ready(best_block_height);
+				match channel_ready_opt {
+					Some(channel_ready) => {
+						channel.unset_requires_manual_readiness_signal();
```

Comment:
Hm yeah, that's how I originally had it but this comment https://github.com/lightningdevkit/rust-lightning/pull/2109#discussion_r1142544337 made it seem like we wanted to re-generate the PendingChannelReady event on restart if they hadn't signaled yet.

So the flow is that a chain event happens, we generate PendingChannelReady, flip `requires_manual_readiness_signal` to false, then LDK terminates.

On next startup, assuming the flag update was persisted, we will go ahead and automatically send ChannelReady when a block is connected without the user manually signaling it's okay.

So if we only have one flag and don't flip it until they signal then every time a block is found we would keep generating the event.

So one flag tracks whether or not we already emitted the event to prevent duplicates on subsequent chain events and the other is supposed to track whether this channel _still_ requires manual signal from user.

I guess the way I have it currently will also persist the "we emitted" flag and would prevent a re-emission on startup which is why Matt mentioned in-memory only?  

Thoughts? I'm kind of confusing myself again around this logic.